soothsayer/Oracle/OracleVersionRespository.cs

```diff
@@ -122,8 +122,28 @@ public void RemoveVersion(DatabaseVersion version, string schema)
             }
         }
 
-        public void InitialiseVersioningTable(string schema, string tablespace = null)
+        public void EnsureVersioningTableIsInitialised(string schema, string tablespace = null)
         {
+            bool alreadyInitialised = VersionTableExists(schema);
+
+            if ( !alreadyInitialised )
+                InitialiseVersioningTable(schema, tablespace);
+        }
+
+        public bool VersionTableExists(string schema)
+        {
```

Comment:
Originally I tried to just get the latest version, and assume the table doesn't exist if the latest version is null.
But that doesn't work if the table exists but is empty. That could happen if the table was created but the user cancelled before actually running any migrations.